### ISSUE TITLE
build: --isolate-devel crashes when not building all packages

### ISSUE THREAD

**Issue by NikolausDemmel** [catkin/catkin_tools#65]
The following failiure occurs in a workspace with more than one package and no existing devel folder (the use of the "--space-suffix" and "--source" options is irrelevant to this issue I assume):

```
$ catkin build --space-suffix test --source srctest --isolate-devel catkin
Creating buildspace directory, '/Users/demmeln/work/indigo_ws/buildtest'
----------------------------------------------------------------------
Workspace:                   /Users/demmeln/work/indigo_ws
Buildspace:                  /Users/demmeln/work/indigo_ws/buildtest
Develspace:                  /Users/demmeln/work/indigo_ws/develtest
Installspace:                /Users/demmeln/work/indigo_ws/installtest
DESTDIR:                     None
----------------------------------------------------------------------
Isolate Develspaces:         True
Install Packages:            False
Isolate Installs:            False
----------------------------------------------------------------------
Additional CMake Args:       None
Additional Make Args:        None
Additional catkin Make Args: None
----------------------------------------------------------------------
Found '2' packages in 0.0 seconds.
Starting ==> catkin
Finished <== catkin [ 3.3 seconds ]
[build] Runtime: 3.5 seconds                                                                                                                                                                                  [0/0 Active | 1/1 Completed]
Traceback (most recent call last):
  File "/usr/local/bin/catkin", line 9, in <module>
    load_entry_point('catkin-tools==0.0.0', 'console_scripts', 'catkin')()
  File "/Users/demmeln/work/catkin_tools/catkin_tools/commands/catkin.py", line 192, in main
    sys.exit(args.main(args) or 0)
  File "/Users/demmeln/work/catkin_tools/catkin_tools/verbs/catkin_build/cli.py", line 183, in main
    lock_install=not opts.no_install_lock
  File "/Users/demmeln/work/catkin_tools/catkin_tools/verbs/catkin_build/build.py", line 548, in build_isolated_workspace
    _create_unmerged_devel_setup(context)
  File "/Users/demmeln/work/catkin_tools/catkin_tools/verbs/catkin_build/build.py", line 233, in _create_unmerged_devel_setup
    first_source=leaf_sources[0],
IndexError: list index out of range
```

**Comment by jbohren** [comment]
I took a moment to look into this. Now that `catkin build` computes a list of "unbuilt" packages, it makes it easy to avoid the leaf-sources error. See #276 for the patch.

**Comment by jbohren** [comment]
Closed with #276
